app/components/community_activity_component/community_activity_component.html.erb

```diff
@@ -14,31 +14,38 @@
-              <span class="govuk-form-group ihavedonethis__group">
-                <%= label_tag "#{:self_verification_info.to_s}_#{@activity.id}",
-                  @activity.self_verification_info&.concat(@activity.uploadable? ? " (#{t('.supported_file_types')})" : ""),
-                  class: 'govuk-label ihavedonethis__text'
-                %>
-                <% if @activity.uploadable? %>
-                  <div class="ihavedonethis__upload-container">
-                    <%= file_field :achievement, :supporting_evidence, class: 'govuk-input' %>
-                  </div>
-                <% else %>
-                  <%= text_field_tag "#{:self_verification_info.to_s}_#{@activity.id}", '', class: 'govuk-input', name: :self_verification_info %>
-                <% end %>
-                <%= f.hidden_field :activity_id, value: @activity.id %>
-              </span>
-              <%= f.submit t('.evidence_submit_label'), class: 'govuk-button button button--focus-white' %>
-            <% end %>
+      <% else %>
+        <% if @activity.self_verification_info.blank? %>
+          <%= form_for Achievement.new, url: achievements_path do |f| %>
+            <%= f.hidden_field :activity_id, value: @activity.id %>
+            <%= f.submit t('.mark_complete_label'), class: 'govuk-button button' %>
+          <% end %>
+        <% else %>
+          <span class="ihavedonethis">
+            <button aria-expanded="true" class="govuk-button button ihavedonethis__button">
+              <%= t('.evidence_button_label') %>
+            </button>
+            <span class="ihavedonethis__popup">
+              <%= form_for Achievement.new, url: achievements_path, :html => { :multipart => true } do |f| %>
+                <span class="govuk-form-group ihavedonethis__group">
+                  <%= label_tag "#{:self_verification_info.to_s}_#{@activity.id}",
+                    @activity.self_verification_info&.concat(@activity.uploadable? ? " (#{t('.supported_file_types')})" : ""),
+                    class: 'govuk-label ihavedonethis__text'
+                  %>
+                  <% if @activity.uploadable? %>
+                    <div class="ihavedonethis__upload-container">
+                      <%= file_field :achievement, :supporting_evidence, class: 'govuk-input' %>
+                    </div>
+                  <% else %>
+                    <%= text_field_tag "#{:self_verification_info.to_s}_#{@activity.id}", '', class: 'govuk-input', name: :self_verification_info %>
+                  <% end %>
+                  <%= f.hidden_field :activity_id, value: @activity.id %>
+                </span>
+                <%= f.submit t('.evidence_submit_label'), class: 'govuk-button button button--focus-white' %>
+              <% end %>
+            </span>
           </span>
-        </span>
+        <% end %>
```

Comment:
The logic here might be easier for a maintainer to follow if you extract the branch bodies as 2 partials: _verify_achievement_without_evidence _verify_achievement_with_evidence .

Reply:
You're right, with my new condition, that became quite the chunky partial. There is no good way to have partials for a view component so I've refactored those if-else statements so its less indented and makes the control flow a little clearer to understand. I hope this is a good compromise